Geometry_Engine/Compute/SortAlongCurve.cs

```diff
@@ -69,6 +83,42 @@ public static List<Point> SortAlongCurve(this List<Point> points, Circle circle,
 
         /***************************************************/
 
+        [Description("Sorts a set of points along the provided curve. Sorting is done based on the curve parameter at the position of the closest point on the curve in relation to each provided Point.")]
+        [Input("points", "The list of points to sort.")]
+        [Input("ellipse", "The Ellipse to sort along.")]
+        [Input("distanceTolerance", "Tolerance to be used for distance computations in the method.", typeof(Length))]
+        [Input("angleTolerance", "Tolerance to be used for angle computations in the methods.", typeof(Angle))]
+        [Output("sortedPts", "The provided points sorted along the curve.")]
+        public static List<Point> SortAlongCurve(this List<Point> points, Ellipse ellipse, double distanceTolerance = Tolerance.Distance, double angleTolerance = Tolerance.Angle)
+        {
+            if(points == null || points.Count == 0)
+                return new List<Point>();
+
+            if (ellipse.IsNull())
+                return new List<Point>();
+
+            if (ellipse.Radius1 <= distanceTolerance && ellipse.Radius2 <= distanceTolerance)
+                return points.Select(p => p.DeepClone()).ToList();
```

Comment:
Thought about that as well, and generally agree. For this first run I basically just did what the other methods did, even if I kind of thought it was unnecessary. 

Happy to remove for this method, but slightly hesitant to do it on the others, just in case there are some weird edge case where another method somewhere assumes that the returned points will be new points, and free to modify. Alternatively we can keep the deep clone on this method, for now, and then remove in separate run, while resolving https://github.com/BHoM/BHoM_Engine/issues/943 .

What do you think?

Reply:
Makes sense, this is exactly what I thought when reading the code. I think it is fine to leave it as is for now, no harm other than performance in the end